File: TaskMaster/AppGlobals/AppStagingFilenames.cs
using TaskMaster.Properties;
using UtilitiesCS;

namespace TaskMaster
{

    public class AppStagingFilenames : IAppStagingFilenames
    {
        private string _conditionalReminders;
        public string ConditionalReminders
        {
            get => _conditionalReminders ?? InitProp(ref _conditionalReminders, Settings.Default.File_ConditionalReminders);
            set
            {
                _conditionalReminders = value;
                Settings.Default.File_ConditionalReminders = value;
                Settings.Default.Save();
            }
        }

        private string _commonWords;
        public string CommonWords
        {
            get => _commonWords ?? InitProp(ref _commonWords, Settings.Default.File_Common_Words);
            set
            {
                _commonWords = value;
                Settings.Default.File_Common_Words = value;
                Settings.Default.Save();
            }
        }

        private string _subjectMap;
        public string SubjectMap
        {
            get => _subjectMap ?? InitProp(ref _subjectMap, Settings.Default.File_Subject_Map);
            set
            {
                _subjectMap = value;
                Settings.Default.File_Subject_Map = value;
                Settings.Default.Save();
            }
        }

        private string _ctfInc;
        public string CtfInc
        {
            get => _ctfInc ?? InitProp(ref _ctfInc, Settings.Default.File_CTF_Inc);
            set
            {
                _ctfInc = value;
                Settings.Default.File_CTF_Inc = value;
                Settings.Default.Save();
            }
        }

        private string _ctfMap;
        public string CtfMap
        {
            get => _ctfMap ?? InitProp(ref _ctfMap, Settings.Default.File_CTF_Map);
            
            set
            {
                _ctfMap = value;
                Settings.Default.File_CTF_Map = value;
                Settings.Default.Save();
            }
        }

        private string _emailSessionTemp;
        public string EmailSessionTemp
        {
            get => _emailSessionTemp ?? InitProp(ref _emailSessionTemp, Settings.Default.FileName_EmailSessionTmp);
            
            set
            {
                _emailSessionTemp = value;
                Settings.Default.FileName_EmailSessionTmp = value;
                Settings.Default.Save();
            }
        }

        private string _emailSession;
        public string EmailSession
        {
            get => _emailSession ?? InitProp(ref _emailSession, Settings.Default.FileName_EmailSession);
            set
            {
                _emailSession = value;
                Settings.Default.FileName_EmailSession = value;
                Settings.Default.Save();
            }
        }

        private string _movedMails;
        public string MovedMails
        {
            get => _movedMails ?? InitProp(ref _movedMails, Settings.Default.FileName_MovedEmailsBackup);
            set
            {
                _movedMails = value;
                Settings.Default.FileName_MovedEmailsBackup = value;
                Settings.Default.Save();
            }
        }

        private string _recentsFile;
        public string RecentsFile
        {
            get => _recentsFile ?? InitProp(ref _recentsFile, Settings.Default.FileName_Recents);
            set
            {
                _recentsFile = value;
                Settings.Default.FileName_Recents = value;
                Settings.Default.Save();
            }
        }

        private string _emailInfoStagingFile;
        public string EmailInfoStagingFile { get => _emailInfoStagingFile ?? InitProp(ref _emailInfoStagingFile, Settings.Default.FileName_EmailInfoStaging); set => _emailInfoStagingFile = value;}

        internal string InitProp(ref string prop, string value)
        {
            prop = value;
            return value;
        }
    }
}

File: TaskMasterOld/AppGlobals/AppStagingFilenames.vb
Imports UtilitiesVB

Public Class AppStagingFilenames
    Implements IAppStagingFilenames

    Private _recentsFile As String = "9999999RecentsFile.txt"
    Private _emailMoves As String = "999999EmailMoves.tsv"
    Private _emailSession As String = "99999EmailSession.csv"
    Private _emailSessionTemp As String = "99999EmailSession_Tmp.csv"
    Private _ctfMap As String = "9999999CTF_Map.txt"
    Private _ctfInc As String = "9999999CTF_Inc.txt"
    Private _subjectMap As String = "9999999Subject_Map.txt"
    Private _commonWords As String = "9999999CommonWords.txt"
    Private _conditionalReminders As String = "999999ConditionalReminders.txt"

    Public Property ConditionalReminders As String Implements IAppStagingFilenames.ConditionalReminders
        Get
            Return _conditionalReminders
        End Get
        Set(value As String)
            _conditionalReminders = value
        End Set
    End Property

    Public Property CommonWords As String Implements IAppStagingFilenames.CommonWords
        Get
            Return _commonWords
        End Get
        Set(value As String)
            _commonWords = value
        End Set
    End Property

    Public Property SubjectMap As String Implements IAppStagingFilenames.SubjectMap
        Get
            Return _subjectMap
        End Get
        Set(value As String)
            _subjectMap = value
        End Set
    End Property

    Public Property CtfInc As String Implements IAppStagingFilenames.CtfInc
        Get
            Return _ctfInc
        End Get
        Set(value As String)
            _ctfInc = value
        End Set
    End Property

    Public Property CtfMap As String Implements IAppStagingFilenames.CtfMap
        Get
            Return _ctfMap
        End Get
        Set(value As String)
            _ctfMap = value
        End Set
    End Property

    Public Property EmailSessionTemp As String Implements IAppStagingFilenames.EmailSessionTemp
        Get
            Return _emailSessionTemp
        End Get
        Set(value As String)
            _emailSessionTemp = value
        End Set
    End Property

    Public Property EmailSession As String Implements IAppStagingFilenames.EmailSession
        Get
            Return _emailSession
        End Get
        Set(value As String)
            _emailSession = value
        End Set
    End Property

    Public Property EmailMoves As String Implements IAppStagingFilenames.EmailMoves
        Get
            Return _emailMoves
        End Get
        Set(value As String)
            _emailMoves = value
        End Set
    End Property

    Public Property RecentsFile As String Implements IAppStagingFilenames.RecentsFile
        Get
            Return _recentsFile
        End Get
        Set(value As String)
            _recentsFile = value
        End Set
    End Property
End Class
